### PR TITLE
Make Vite the default bundler

### PR DESCRIPTION
**Pull request by dac09**
This PR does the following:
- makes Vite the default bundler
- updates CRWA TS template to have Vite setup
- updates CRWA JS template to have Vite setup
- rebuilds the fixture Vite preconfigured
- Vite specific files are now accessible via `getPaths` in `rw/project-config`
- ⚠️ breaks SVG handling, so it now works in the default Vite way instead of using svg components (which is fragile and confusing to users, see https://github.com/redwoodjs/redwood/issues/8291) 

**Todo**
- [x] Unable to use the vite dev server in Gitpod for reviewing PRs
~This probably is just a small config tweak, but not sure what it is.~

- [x] Failing Windows smoke test with Playwright
~This is likely the same issue, accessing the port via Playwright.~

### CODE REVIEW

File: .gitpod.yml [hunk 1]
```diff
@@ -23,9 +23,10 @@ tasks:
       cd /workspace/redwood
       yarn install
       yarn run build:test-project ../rw-test-app --typescript --link --verbose
+      cd /workspace/rw-test-app && sed -i "s/\(open *= *\).*/\1false/" redwood.toml
```
Comment (dac09) [inline .gitpod.yml:26]: The open flag does not work in Gitpod

File: packages/vite/bins/rw-vite-dev.mjs [hunk 2]
```diff
@@ -11,9 +12,25 @@ const startDevServer = async () => {
     throw new Error('Could not locate your web/vite.config.{js,ts} file')
   }
 
+  // Tries to maintain the same options as vite's dev cli
+  // See here: https://github.com/vitejs/vite/blob/main/packages/vite/src/node/cli.ts#L103
+  // e.g. yarn rw dev web --fwd="--force"
```
Comment (dac09) [inline packages/vite/bins/rw-vite-dev.mjs:17]: Thoughts on whether `--fwd` still makes sense? 

When we run `yarn rw dev` it runs both, so maybe the flag should be `webFwd`?

File: packages/project-config/src/paths.ts [hunk 3]
```diff
@@ -213,11 +217,37 @@ export const getPaths = (BASE_DIR: string = getBaseDir()): Paths => {
...
+    .sync('*.routeHooks.{js,ts,tsx,jsx}', {
+      absolute: true,
+      cwd: path.dirname(pagePath), // the page's folder
+    })
+    .at(0)
+}
+
+export const getAppRouteHook = () => {
+  return resolveFile(path.join(getPaths().web.src, 'App.routeHooks'))
+}
```
Comment (jtoar) [inline packages/project-config/src/paths.ts:244]: I couldn't find these new functions being used anywhere (just did a simple search); what's the plan for them?

Comment (dac09) [reply packages/project-config/src/paths.ts:244]: I can remove these. They're used in the experimental branch. I just checked out stuff from there.

**Comment by dac09** [comment]
Thank you for your help with this @jtoar - no way I would've solved it without your help 🤾

File: tasks/smoke-test/playwright-fixtures/devServer.fixture.ts [hunk 4]
```diff
@@ -47,50 +49,56 @@ const test = base.extend<any, DevServerFixtures>({
         )
       }
 
-      const isServerAlreadyUp = await isPortReachable(webServerPort, {
-        timeout: 5000,
+      // @MARK: On second run, Web server is off, but api server is on. Why?!
```
Comment (Tobbe) [inline tasks/smoke-test/playwright-fixtures/devServer.fixture.ts:52]: Is this change still needed with @jtoar's CI fix?
Or have we at least gotten a better understanding of this, so we can have a more educating comment here?

File: .github/actions/setup_test_project/setup_test_project.mjs [hunk 5]
```diff
@@ -1,15 +1,20 @@
-import os from 'node:os'
+// import os from 'node:os'
```
Comment (Tobbe) [inline .github/actions/setup_test_project/setup_test_project.mjs:1]: Remove commented code

File: .github/actions/setup_test_project/setup_test_project.mjs [hunk 6]
```diff
@@ -1,15 +1,20 @@
...
 import { exec } from '@actions/exec'
 import * as core from '@actions/core'
 
+// const test_project_path = path.join(
+//   os.tmpdir(),
+//   'test-project',
+//   // ":" is problematic with paths
+//   new Date().toISOString().split(':').join('-')
+// )
+
```
Comment (Tobbe) [inline .github/actions/setup_test_project/setup_test_project.mjs:14]: Remove commented code

File: packages/vite/bins/rw-vite-dev.mjs [hunk 7]
```diff
@@ -26,6 +44,12 @@ const startDevServer = async () => {
   })
 
   devServer.printUrls()
+
+  if(debug) {
```
Comment (Tobbe) [inline packages/vite/bins/rw-vite-dev.mjs:48]:
```suggestion
  if (debug) {
```

**Review by Tobbe — APPROVED** [review]

File: tasks/smoke-test/playwright-fixtures/devServer.fixture.ts (hunk 4 above)
Comment (dac09) [reply tasks/smoke-test/playwright-fixtures/devServer.fixture.ts]: Yeah this was for me during debugging. Removed it now.

**Pull request merged**